File: src/main/java/ru/ioleksiv/telegram/bot/core/model/telegram/method/group/SetChatDescription.java
package ru.ioleksiv.telegram.bot.core.model.telegram.method.group;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.ioleksiv.telegram.bot.core.controller.network.Networker;
import ru.ioleksiv.telegram.bot.core.model.telegram.method.interfaces.ChatAction;
import ru.ioleksiv.telegram.bot.core.model.telegram.responses.CommonResponse;
import ru.ioleksiv.telegram.bot.core.model.telegram.responses.ResponseCollection;

/**
 * @see <a href="https://core.telegram.org/bots/api#setchatdescription>setChatDescription</a>
 */
public class SetChatDescription  extends ChatAction<Boolean > {
    private static final String METHOD = "setChatDescription";
    /**
     * description	String	Optional	New chat description, 0-255 characters
     */
    @JsonProperty("description")
    private String description;

    protected SetChatDescription(Networker networker) {
        super(METHOD, networker);
    }

    public String getDescription() {
        return description;
    }

    public SetChatDescription  setDescription(String description) {
        this.description = description;
        return this;
    }

    @Override
    public Class<? extends CommonResponse<Boolean>> getResultWrapperClass() {
        return ResponseCollection.BooleanResponse.class;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/method/group/SetChatDescription.kt
package io.github.oleksivio.tl.kbot.server.api.method.group

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.ApiDict
import io.github.oleksivio.tl.kbot.server.api.model.BooleanResponse
import io.github.oleksivio.tl.kbot.server.api.model.method.common.ChatAction
import io.github.oleksivio.tl.kbot.server.api.objects.ChatId

/**
 * @see [setChatDescription](https://core.telegram.org/bots/api/#setchatdescription)
 */
data class SetChatDescription(
    /**
     * chat_id Integer or String Yes Unique identifier for the target chat or username of the target channel
     */
    @JsonProperty(ApiDict.CHAT_ID_KEY)
    override val chatId: ChatId,
    /**
     * description [String] Optional New chat description, 0-255 characters
     */
    @JsonProperty(ApiDict.DESCRIPTION_KEY)
    val description: String? = null
) : ChatAction<Boolean>() {

    @JsonIgnore
    override val resultWrapperClass = BooleanResponse::class

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "setChatDescription"
}

